### ISSUE TITLE
Code Lists - Link to external ontologies

### ISSUE THREAD

**Issue by lambertciata** [IATA-Cargo/ONE-Record#225]
Some Code Lists could be linked to external ontologies such as UNCEFACT public vocabulary.

See: https://vocabulary.uncefact.org/ and https://vocabulary.uncefact.org/PartyRoleCodeList for an example.

**Comment by lambertciata** [comment]
On the principle I think this is a no brainer, however a full review of impacted code lists is necessary. I suggest to do that in the next couple of months in order to aim for the next release (end of 2024)

**Comment by lambertciata** [comment]
Example given: Measurement units, Country codes?

**Comment by lambertciata** [comment]
We agreed to have a review of Code Lists and identify the ones where external ontology apply.

**Comment by lambertciata** [comment]
Quick update on this after discussing with the team responsible of the vocabulary at UN/CEFACT: As of today it is not really maintained, there were discussions about updating the vocabulary but it is not clear.

My proposal as of today: We can indicate reference to existing external ontologies such as UN/CEFACT vocabulary but we should keep ours as it would be maintained through 1R standard. We know that lists such as measurements and units will not evolve.

**Comment by ddoeppner** [comment]
With regard to the shipping record – see xFWB mapping to a record – 
what is the final, authoritative guideline for mapping Currency and country?

Country examples from xFWB
```xml
<ram:IncludedCustomsNote> 	
	<ram:CountryID>XXXXXXXXXX</ram:CountryID>
</ram:IncludedCustomsNote>	
```

```xml
<ram:PostalStructuredAddress> 	
	<ram:CountryID>DE</ram:CountryID>
</ram:PostalStructuredAddress>	
```

will this map to
```json
{
  "@type": "cargo:Address",
  "cargo:country": {
    "@type": "cargo:CodeListElement",
    "cargo:code": "DE"
  }
}
```

or

```json
{
  "@type": "cargo:Address",
  "cargo:country": {
    "@id": "https://vocabulary.uncefact.org/CountryId#DE"
  }
}
```

similar question wrt. Currency and usage of https://vocabulary.uncefact.org/AmountCurrency instead of CodeListElement

```xml
<CarrierTotalDuePayableAmount currencyID="EUR">4711</CarrierTotalDuePayableAmount>
```

```json
"cargo:currencyUnit": {
          "@type": "cargo:CodeListElement",
          "cargo:code": "EUR"
}
```

or
```json
"cargo:currencyUnit": {
          "@id": "https://vocabulary.uncefact.org/AmountCurrency#EUR"
}
```

Thanks for guidance

**Comment by nscheiber-champ** [comment]
Specifications 2025-07 are clear here:
https://iata-cargo.github.io/ONE-Record/2025-07/Data-Model/code-lists/

Any CodeListElement **MUST** have the code as part of its URI:
` @id: {{baseURL}}/{{codeListName}}#{{code}}`

The URI does MAY not be resolvable. The properties defined in cargo:CodeListElement can be used if the URI is resolvable.

We will add an OWL axiom to the ontology with the next release.

**Comment by ddoeppner** [comment]
thank you for clarification, @nscheiber-champ 
back to my question: Should we use  `CodeListElement` or UNEFACT vocab for currencyID and countryID used in cXML, like xFWB?

**Comment by nscheiber-champ** [comment]
This is not specified. CXML tends to use ISO codes over UNCEFACT, though they mostly overlap. In EDI and paper AWB, a mixture of everything is in use.

Both is permitted, it is just important that in the `@id` you include the source of the code list `{{baseURL}}/{{codeListName}}` and have a fragment that is an element of the code list `#{{code}}`.

If you believe we need to specify it further, we can discuss this in the CXML mapping task force.

**Comment by ddoeppner** [comment]
I agree. For the foundational use case `Shipment Record` and its mapping to/from cXML, it would be beneficial to define the code lists to be used as explicitly and concretely as possible. Clear, standardized code lists help minimize ambiguity and reduce the risk of fragmented, inconsistent, or incompatible implementations of the shipment record over time.

**Comment by nscheiber-champ** [comment]
Unless specified otherwise in Resolution 600a, ONE Record must handle all. I ask IATA to validate please @lambertciata 

If that is the case, for Shipment Record, the only thing we can do is provide a recommendation. As a carrier you can always mandate.

**Comment by lambertciata** [comment]
Hello a few elements here:

- CXML is based on ISO Currency and Country codes, they are hardcoded in the schemas ... this means that whenever there is an update of those code lists, XML schemas need to be updated with a new version.
- Most Resolutions indicate the ISO Currency and Country code lists as the source. We need to stick to that as a requirement for 1R. There are political reasons behind this choice.
- Since both lists are public, I guess we could publish the code lists as part of the ontology, however it implies that we need to maintain it and potentially hotfix when there are changes
